ttnn/cpp/ttnn/compiler_interface/compiler_interface.hpp

```diff
@@ -0,0 +1,80 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <optional>
+#include <string>
+
+#include "third_party/json/json.hpp"
+#include "ttnn/graph/graph_trace_utils.hpp"
+
+namespace ttnn::compiler_interface {
```

Comment:
I'm unsure this is a good naming. "Compiler interface" typically means "Interface **of** Compiler", not "Interface **for** compiler". I would rather call it just `op_constraints`.

Reply:
if you and Artem are aligned on the new name - I'll make a change